workspaces/javascript-leetcode-month/problems/2703-return-length-of-arguments-passed/solution.md

```diff
@@ -0,0 +1,314 @@
+
+[View submission on LeetCode](https://leetcode.com/problems/return-length-of-arguments-passed/submissions/1374678620/)
+
+```ts
+const argumentsLength = (...args: readonly unknown[]): number => args.length;
+
+/**
+ * argumentsLength(1, 2, 3); // 3
+ */
+```
+
+Note that TypeScript type annotations are often optional, if there is enough context to infer a type. For example, the return type of the function can be inferred to be a number, because we are returning the `length` of an array, which is a number:
+
+[View submission on LeetCode](https://leetcode.com/problems/return-length-of-arguments-passed/submissions/1374678865/)
+
+```ts
+const argumentsLength = (...args: readonly unknown[]) => args.length;
+
+/**
+ * argumentsLength(1, 2, 3); // 3
+ */
+```
+
+If you are new to TypeScript, I recommend using type annotations as much as possible to get the practice. In the future, we'll discuss where it's safe to omit them without sacrificing safety.
+
+### Using `arguments`
+
+If we use the `arguments` object, the rest parameter becomes unused, so we can choose to not even declare it!
+
+[View submission on LeetCode](https://leetcode.com/problems/return-length-of-arguments-passed/submissions/1374679390/)
+
+```js
+/**
+ * @return {number}
+ */
+function argumentsLength() {
+  return arguments.length;
+}
+
+/**
+ * argumentsLength(1, 2, 3); // 3
+ */
+```
+
+However, in TypeScript, omitting the rest parameter will result in a type error in the code LeetCode uses to invoke our function! A common practice is to prefix the names of unused variables with an underscore:
+
+[View submission on LeetCode](https://leetcode.com/problems/return-length-of-arguments-passed/submissions/1374680348/)
+
+```ts
+function argumentsLength(..._args: readonly unknown[]): number {
+  return arguments.length;
+}
+
+/**
+ * argumentsLength(1, 2, 3); // 3
+ */
+```
+
+We can also do something a bit funny-looking, and use an empty destructure of the rest parameter! Destructuring of the rest parameter is documented in [the rest parameters documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters).
+
+[View submission on LeetCode](https://leetcode.com/problems/return-length-of-arguments-passed/submissions/1374680779/)
+
+```ts
+function argumentsLength(...[]: readonly unknown[]): number {
+  return arguments.length;
+}
+
+/**
+ * argumentsLength(1, 2, 3); // 3
+ */
+```
+
+Fun fact, `arguments` don't work the same way in arrow functions. [The documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) phrases it:
+
+> Arrow functions don't have their own bindings to `this`, `arguments`, or `super`
```

Comment:
ah thanks for the reminder, I was wondering why using `arguments` wasn't working with arrow functions